pkg/mutation/match/match.go

```diff
@@ -47,7 +47,7 @@ type Kinds struct {
 
 // Matches verifies if the given object belonging to the given namespace
 // matches the current mutator.
-func Matches(match Match, obj runtime.Object, ns *corev1.Namespace) (bool, error) {
+func Matches(match *Match, obj runtime.Object, ns *corev1.Namespace) (bool, error) {
```

Comment:
fwiw `Match` is just a bunch of pointers and length integers. I'm not sure at what threshold gocritic defines something as "big". I did a microbenchmark and copying vs taking a pointer of this struct and the difference is negligible.
I don't have an objection to the change, and I like to nerd-out on microbenchmarks - overall we should probably apply metrics and benchmarks when evaluating optimizations.

Reply:
The linter default is 80 bytes, which is just at the threshold where most use cases see a minuscule gain when using pointers instead of structs. I'm happy to change this setting - many repos use 100 or 200: preferring to only have linter failures in cases where there's certainly a gain.

Reply:
I don't know what the right magic number is - I tend to agree with Peter Bourgon's advice on this: _you should always make the "pointer or value?" decision based on required semantics, i.e. "mutate or not?", until profiling proves otherwise._

We're in micro optimization territory here so I'd always opt for code clarity, safety, etc unless there's a reason not to.

Reply:
We can also just disable that specific sub-linter.